lms/djangoapps/student_account/views.py

```diff
@@ -123,6 +124,17 @@ def login_and_registration_form(request, initial_mode="login"):
         } for message in messages.get_messages(request) if 'account-activation' in message.tags
     ]
 
+    account_creation_allowed = configuration_helpers.get_value(
+        'ALLOW_PUBLIC_ACCOUNT_CREATION', settings.FEATURES.get('ALLOW_PUBLIC_ACCOUNT_CREATION', True)
+    )
+
+    running_pipeline = pipeline.get(request)
+    if running_pipeline:
+        if user_exists(running_pipeline['kwargs']['details']):
+            # Details sent over from identity provider belong to an existing user.
+            # So, force user to login and do not allow account creation
+            account_creation_allowed = False
```

Comment:
But we only want to force login if a user account exists AND `sync_learner_profile_data` is `True`. I think you want to keep a separate boolean in the third_party_auth context and check it in the front-end code. `account_creation_allowed` serves a slightly different purpose.

Reply:
Done